File: Artificial Intelligence/SumNTerms.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jan 22 14:29:48 2021

@author: Dragneel
"""

#%% The recursive function
def sumSeries(start, n, inv):
    if n == 1:
        return start
    return start + (n-1)*inv + sumSeries(start, n-1, inv)

#%% Input Handling and Calling recursive Function

print('Printing the sum of First N terms of an equal interval Series')

start = int(input('Enter the first element of series: '))
n = int(input('Number of Elements to Calculate the Sum: '))
inv = int(input('Interval among consecutive numbers: '))

ans = sumSeries(start, n, inv)

print('The sum is: ', ans)

File: Artificial Intelligence/SumNTerms.pl
sumSeries(1, S, _, S) :- !.
sumSeries(N, S, I, V) :- N1 is N-1, sumSeries(N1, S, I, S1), V is S1+ S + N1*I.

seriesEntry :- write('Starting Number: '), read(A),
               write('Number of Elements: '), read(B),
               write('Interval: '), read(C), write('The sum is: '),
               sumSeries(B, A, C, X), write(X).
